src/System.Linq/src/System/Linq/Enumerable.cs

```diff
@@ -1610,10 +1614,61 @@ public static int Sum(this IEnumerable<int> source)
             int sum = 0;
             checked
             {
-                foreach (int v in source) sum += v;
+                if (Vector.IsHardwareAccelerated && source.Count() >= SIMD_THRESHOLD)
```

Comment:
Potentially the source may not be replayable, and so it would end up with `Sum` throwing an exception or returning `0` because on the subsequent attempt there were no elements.